tests/unit/aspect.ts

```diff
@@ -171,6 +171,41 @@ registerSuite({
 			}
 		},
 
+		'.on()': {
+			'advising function returns undefined, returns original result'() {
+				const aspectStub1 = sinon.stub();
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 1);
+				assert.isTrue(aspectStub1.calledAfter(methodSpy));
+			},
+
+			'advising function returns defined values, returns advising function result'() {
+				const aspectStub1 = sinon.stub().returns(2);
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 2);
```

Comment:
* [Spring AOP](http://docs.spring.io/spring/docs/current/spring-framework-reference/html/aop.html)
* [AspectJ](https://eclipse.org/aspectj/doc/next/progguide/semantics-advice.html)
* [Go AOP](http://go.aopphp.com/docs/pointcuts-and-advices/) this one doesn't have *After Finally* advice

I might be interesting to deliver some of these other types of advice (*After Finally* and *After Throwing*).